Source/Particles/ShapeFactors.H

```diff
@@ -89,30 +89,80 @@ struct Compute_shifted_shape_factor
     {
         if constexpr (depos_order == 1){
             const auto i = static_cast<int>(x_old);
-            const int i_shift = i - i_new;
+            //const int i_shift = i - i_new;
+            const bool shift = i != i_new;
             const T xint = x_old - T(i);
-            sx[1+i_shift] = T(1.0) - xint;
-            sx[2+i_shift] = xint;
+
+            //amrex::AllPrint() << "i_shift=" << i_shift << "\n";
+
+            // we avoid dynamic indexing here, to enable vectorization (CPU) and
+            // reduce register/lmem footprint (GPU).
+            // The below is equivalent to this:
+            //sx[1+i_shift] = T(1.0) - xint;
+            //sx[2+i_shift] = xint;
+            sx[1] = T(1.0) - xint;
+            sx[2] = xint;
+
+            sx[0] = shift ? sx[1] : T(0.0);
```

Comment:
Maybe I'm missing something, but this only seems correct if the old `i_shift` is -1 or 0. What happens if `i_shift` is +1?

Reply:
According to a quick test, it can only take the values 0 and -1 in these locations.
CI will tell.

Reply:
Interesting, seems there is more to it here. We should really change those ranges to simplify this, I think there is a lot of zero-padding going on here.

Reply:
That you are not moving in the negative direction is only true if you check the position of the start and end particle trajectory. You must use the lower value as a base for your coordinate system (to calculate the shift), then the particle trajectory is always crossing the positive side of a cell.